File: lismove-iOS/LisMove/Modules/Shared/Model/UserDashboard.swift
//
//  UserDashboard.swift
//  LisMove
//
//

import Foundation
public class UserDashboard: Codable{
    var co2: Double?
    var distance: Double?
    var euro: Double?
    var messages: Int?
    var sessionNumber: Int?
    var dailyDistance: [UserDistanceStats]?
    
    enum CodingKeys: String, CodingKey{
        case co2 = "co2"
        case distance = "distance"
        case euro = "euro"
        case messages = "messages"
        case sessionNumber = "sessionNumber"
        case dailyDistance = "dailyDistance"
    }
    
    public init(co2: Double?, distance: Double?, euro: Double?, messages: Int?, sessionNumber: Int?, dailyDistance: [UserDistanceStats]?){
        self.co2 = co2 ?? 0.0
        self.distance = distance ?? 0.0
        self.euro = distance ?? 0.0
        self.messages = messages ?? 0
        self.sessionNumber = sessionNumber ?? 0
        self.dailyDistance = dailyDistance ?? [UserDistanceStats]()
    }
    
    func getConvertedC02() -> Double{
        if let co2 = co2 {
            return co2 / 1000
        }else{
            return 0.0
        }
    }
    
}

public class UserDistanceStats: Codable{
    var day: String?
    var distance: Double?
    
    enum CodingKeys: String, CodingKey{
        case day = "day"
        case distance = "distance"
    }
    
    public init(day: String?, distance: Double?){
        self.day = day
        self.distance = distance
    }
    
    public func getDayString()-> String{
        return DateTimeUtils.getReadableMonthYear(dateTime: self.day ?? "")
    }
}

File: lismove-backend/src/main/java/net/nextome/lismove/rest/dto/UserDashboard.java
package net.nextome.lismove.rest.dto;

import net.nextome.lismove.models.query.UserDistanceStats;

import java.util.List;

public class UserDashboard {

	private Integer sessionNumber;
	private Double sessionDistanceAvg;
	private Double distance;
	private Double co2;
	private Double euro;
	private List<UserDistanceStats> dailyDistance;
	private Integer messages;

	public Integer getSessionNumber() {
		return sessionNumber;
	}

	public void setSessionNumber(Integer sessionNumber) {
		this.sessionNumber = sessionNumber;
	}

	public Double getSessionDistanceAvg() {
		return sessionDistanceAvg;
	}

	public void setSessionDistanceAvg(Double sessionDistanceAvg) {
		this.sessionDistanceAvg = sessionDistanceAvg;
	}

	public Double getDistance() {
		return distance;
	}

	public void setDistance(Double distance) {
		this.distance = distance;
	}

	public Double getCo2() {
		return co2;
	}

	public void setCo2(Double co2) {
		this.co2 = co2;
	}

	public Double getEuro() {
		return euro;
	}

	public void setEuro(Double euro) {
		this.euro = euro;
	}

	public List<UserDistanceStats> getDailyDistance() {
		return dailyDistance;
	}

	public void setDailyDistance(List<UserDistanceStats> dailyDistance) {
		this.dailyDistance = dailyDistance;
	}

	public Integer getMessages() {
		return messages;
	}

	public void setMessages(Integer messages) {
		this.messages = messages;
	}
}
